### DISCUSSION TITLE
How to updade repo wiki from Github Actions

### DISCUSSION THREAD

**Discussion by JacekPliszka** [community/community#25929] [Actions]
<p>Is there an easy way to update repo wiki from Github Actions?</p>
<p>I’ve tried cloning it but it does not work:</p>
<pre><code class="lang-auto">        export URL=ssh://${{ github.token }}@github.com/${{ github.repository }}.wiki
        git clone $URL ${BUILD_DIR}/markdown
</code></pre>
<p>I am getting</p>
<p>Warning: Permanently added the RSA host key for IP address ‘140.82.114.4’ to the list of known hosts.</p>
<p>***<span class="mention">@github.com</span>: Permission denied (publickey).<br>
fatal: Could not read from remote repository.</p>
<p>with https I am getting<br>
Cloning into ‘/home/runner/work/_temp/build/markdown’…</p>
<p>fatal: could not read Password for ‘<a>https://***@github.com</a>’: No such device or address</p>
<p>Is there an easy way to get wiki related to the repo ?</p>
<p>E.g. when one checks out repo, could there be an option to checkout wiki too ?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jacekpliszka">@JacekPliszka</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>Github repo wiki is actually a repo with named reponame.wiki, hence you can specify the wiki repo name to actions/checkout to checkout it. Code sample as below:</p>
<pre><code class="lang-auto">      - uses: actions/checkout@v2
        with:
          repository: ${{github.repository}}.wiki    # specify the wiki repo name here.
        
      - name: check wiki repo content
        run: |
          ls -al
      - name: edit wiki
        run: |
          echo test &gt;&gt; test.md
      - name: Commit files
        run: |
          git config --local user.email "action@github.com"
          git config --local user.name "GitHub Action"
          git add .
          git commit -m "Add changes"
      - name: Push changes to wiki repo
        uses: ad-m/github-push-action@master
        with:
          repository: ${{github.repository}}.wiki    # specify the wiki repo and push the update.
          github_token: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>Please refer to my <a href="https://github.com/weide-zhou/ticket13/runs/840021192?check_suite_focus=true" rel="nofollow noopener">workflow</a> for more details.</p>
<p>BTW, you can also search existing actions in markeplace <a href="https://github.com/marketplace?type=actions&amp;query=wiki" rel="nofollow noopener">here</a>, for example, use action ‘SwiftDocOrg/github-wiki-publish-action@v1’ to publish content to wiki.</p>
<p>Hope it clear and helpful!</p>

**Comment by JacekPliszka** [comment]
<p>Thank you, I should’ve specified a bit more what I needed but with your suggestion it worked great.</p>
<p>What I needed is to have both base repo and wiki at the same time. But in addition to repository parameter there is path parameter so everything works great now.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jacekpliszka">@JacekPliszka</a>,</p>
<p>Thanks for your reply! Yes, you can use path parameter of actions/checkout to store base repo and wiki repo seperately. Glad to know it works for you and issue is gone.</p>
<p>Happy coding!</p>

**Comment by Andre601** [comment]
<p>If I now understand this right, should the <code>GITHUB_TOKEN</code> have write access to the wiki, so that I could f.e. define it as the <code>GH_PAT</code> enviroment for the Decathlon/wiki-page-creator-action?</p>

**Comment by JacekPliszka** [comment, marked as answer]
<p>I did not have to do anything - all I did was</p>
<pre><code class="lang-auto">    - name: Checkout base code
      uses: actions/checkout@v2
      with:
        path: base-code
    - name: Checkout wiki code
      uses: actions/checkout@v2
      with:
        repository: ${{github.repository}}.wiki
        path: markdown
    - name: Push to wiki
      run: |
        cd markdown
        git config --local user.email "action@github.com"
        git config --local user.name "GitHub Action"
        git add .
        git diff-index --quiet HEAD || git commit -m "Add changes" &amp;&amp; git push</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/Andre601">@Andre601</a>,</p>
<p>Yes, <code>GITHUB_TOKEN</code> has write access to the wiki. <code>Decathlon/wiki-page-creator-action</code>  is a third-party action, you can use it with token to <code>publish</code> wikipages to the repository.</p>
<p>Hi <a class="mention" href="/jacekpliszka">@JacekPliszka</a>,</p>
<p>Since the issue is resloved, could you please help to mark the answer?  It could be helpful to other who has same queries. Let me know if you have any questions.</p>
<p>Thanks.</p>

**Comment by Andre601** [comment]
<p>good to know,</p>
<p>However do I face the issue, that the action I use doesn’t work with the <code>GITHUB_TOKEN</code> secret, either because of how the action handles git actions or because of the repo being part of an organisation.<br>
Do orgs have a different behaviour for wikis or <em>should</em> the same action work?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/Andre601">@Andre601</a>,</p>
<p>The action <code>Decathlon/wiki-page-creator-action</code>  seems to handle the access as a remote repository which ‘GITHUB_TOKEN’ doesn’t work, have to use PAT secrets.</p>
<p>Actually in my <a href="https://github.community/t/how-to-updade-repo-wiki-from-github-actions/121151/2">example</a>, you can find GITHUB_TOKEN can push update to wiki repo. You can follow it, submit your changes to wiki.</p>
<p>Thanks.</p>
